staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/converter.go

```diff
@@ -234,9 +233,8 @@ func (c *delegatingCRConverter) ConvertToVersion(in runtime.Object, target runti
 			return in, nil
 		}
 		// for a list, set the version of the top-level list object (all individual objects are already in the correct version)
-		out := list.DeepCopy()
```

Comment:
I'm trying to understand why the deep copy was there and if it's safe to remove... looks like it was original to the converter in https://github.com/kubernetes/kubernetes/pull/67006

I can't tell if this is introducing modification of the input (or if we were already doing that), and if so, if that's ok. I see the existing function docs around converting `in` in-place, but I'm not sure callers actually had that expectation if it wasn't happening previously.

Reply:
the interface does not require the implementation to be safe, see above "Note that the function may mutate in object and return it." We wrap it with another deepcopy lateron.